### ISSUE TITLE
ASP.NET Core 1.0 compatibility

### ISSUE THREAD

**Issue by ArgusDev** [IdentityServer/IdentityServer3#3027]
- [x] I read and understood how to enable [logging](https://identityserver.github.io/Documentation/docsv2/configuration/logging.html)

### Question / Issue

IdentityServer3 worked Ok with ASP.NET 5 RC1 and RC2 however after I upgraded it ASP.NET Core 1.0 RTM it does not work anymore.
It looks like IdentityServer3 looses requests body. First I had the problem exactly how it was raised previously with [AntiForgery validation](https://github.com/IdentityServer/IdentityServer3/issues/1731) that I could bypass by disabling logging. It's reproducible with upgraded (to ASP.NET Core 1.0) [sample](https://github.com/IdentityServer/IdentityServer3.Samples/tree/master/source/AspNet5/src/IdentityServer).
However the problem affects other areas as well. For example during "parsing Basic Authentication secret" it says "No secret in post body found".

### Relevant parts of the log file

```
----------------------------------------
Message: Start client validation
Category: IdentityServer3.Core.Validation.ClientSecretValidator
----------------------------------------
Message: Start parsing Basic Authentication secret
Category: IdentityServer3.Core.Validation.BasicAuthenticationSecretParser
----------------------------------------
Message: Start parsing for secret in post body
Category: IdentityServer3.Core.Validation.PostBodySecretParser
----------------------------------------
Message: No secret in post body found
Category: IdentityServer3.Core.Validation.PostBodySecretParser
----------------------------------------
Message: Start parsing for X.509 certificate
Category: IdentityServer3.Core.Validation.X509CertificateSecretParser
----------------------------------------
Message: client_id is not found in post body
Category: IdentityServer3.Core.Validation.X509CertificateSecretParser
----------------------------------------
Message: Parser found no secret
Category: IdentityServer3.Core.Validation.SecretParser
----------------------------------------
Message: {
  "Category": "ClientAuthentication",
  "Name": "Client authentication failure",
  "EventType": "Failure",
  "Id": 1051,
  "Message": "No client id or secret found",
  "Details": {
    "ClientId": "unknown",
    "ClientType": "Client"
  },
  "Context": {
    "ActivityId": "f906e20f-45f9-4448-82ea-10b85921b41a",
    "TimeStamp": "2016-06-30T01:51:51.149961+00:00",
    "ProcessId": 12784,
    "MachineName": "MyMachine",
    "RemoteIpAddress": "fe80::557d:7cf2:bd52:7cdd%4"
  }
}

Category: Events
----------------------------------------
Message: No client secret found
Category: IdentityServer3.Core.Validation.ClientSecretValidator
----------------------------------------
Message: {
  "Category": "Endpoints",
  "Name": "Endpoint failure",
  "EventType": "Failure",
  "Id": 3001,
  "Message": "invalid_client",
  "Details": {
    "EndpointName": "token"
  },
  "Context": {
    "ActivityId": "f906e20f-45f9-4448-82ea-10b85921b41a",
    "TimeStamp": "2016-06-30T01:51:51.1569609+00:00",
    "ProcessId": 12784,
    "MachineName": "MyMachine",
    "RemoteIpAddress": "fe80::557d:7cf2:bd52:7cdd%4"
  }
}
Category: Events
----------------------------------------
```

What can be done to make it work with ASP.NET Core 1.0?

Thank you very much.

**Comment by leastprivilege** [comment]
@JakeGinnivan do you have the same problems?

**Comment by hiredgunhouse** [comment]
When will you support 1.0 RTM bits? It's been a few days already.
I'm especially interested in IdentityServer3.Integration.AspNetCore cause it references old Microsoft.Framework.DependencyInjection :(

**Comment by leastprivilege** [comment]
feel free to send a PR

**Comment by hiredgunhouse** [comment]
@leastprivilege I do, but does your answer mean that you have no plans to migrate to RTM? I could probably give it a try but I have no expertise in the area of OpenID and OAuth so I'm not sure you would like someone like me mess with your code ;P

**Comment by hiredgunhouse** [comment]
@leastprivilege Now I'm confused, there's IdentityServer4 which is supposed to be for ASP.NET Core 1.0 so is there any point in upgrading IdentityServer3 to ASP.NET Core 1.0? What is the relation of the two projects?

**Comment by leastprivilege** [comment]
@ArgusDev You seem to be the only person with that issue.
@hiredgunhouse IdentityServer4 runs natively on aspnet core. There is also a "compat shim" that allows IdentityServer3 to run on aspnet core. IdentityServer4 is not done yet.

**Comment by hiredgunhouse** [comment]
@leastprivilege looks like in the meantime @IngbertPalm did the work and you pulled it: https://github.com/IdentityServer/IdentityServer3.Integration.AspNetCore/commit/872bc710dcd34d15d8cfd160b4b4a54e285b42f0  
I'll see if that works for me and get back to you here.

**Comment by leastprivilege** [comment]
nuget published

**Comment by hiredgunhouse** [comment]
@leastprivilege thanks, this solved my problem.

**Comment by asteffes** [comment]
I'm still experiencing this problem. 

To recreate it in my solution I can un-comment the EnableHttpLogging:

` LoggingOptions = new LoggingOptions {
                    //EnableHttpLogging = true, //this must remain commented out until we upgrade Core. This will cause anti-forgery token
                    //https://github.com/IdentityServer/IdentityServer3/issues/3027
                    EnableKatanaLogging = true,
                    EnableWebApiDiagnostics = true,
                    WebApiDiagnosticsIsVerbose = true,
                }`


Project.json:

`{
    "buildOptions": {
        "emitEntryPoint": true,
        "preserveCompilationContext": true
    },
    "configurations": {
        "Development": {}
    },
    "dependencies": {
        "Autofac": "4.0.0-rc1-177",
        "Autofac.Extensions.DependencyInjection": "4.0.0-rc1-177",
        "IdentityServer3": "2.1.1",
        "IdentityServer3.Integration.AspNetCore": "1.0.0",
         "Microsoft.ApplicationInsights.AspNetCore": "1.0.0",
        "Microsoft.AspNet.Identity.Core": "2.2.1",
        "Microsoft.AspNet.Identity.EntityFramework": "2.2.1",
        "Microsoft.AspNetCore.Diagnostics": "1.0.0",
        "Microsoft.AspNetCore.Mvc": "1.0.0",
        "Microsoft.AspNetCore.Razor.Tools": {
            "type": "build",
            "version": "1.0.0-preview2-final"
        },
        "Microsoft.AspNetCore.Server.IISIntegration": "1.0.0",
        "Microsoft.AspNetCore.Server.Kestrel": "1.0.0",
        "Microsoft.AspNetCore.StaticFiles": "1.0.0",
        "Microsoft.Extensions.Configuration.EnvironmentVariables": "1.0.0",
        "Microsoft.Extensions.Configuration.Json": "1.0.0",
        "Microsoft.Extensions.Logging": "1.0.0",
        "Microsoft.Extensions.Logging.Console": "1.0.0",
        "Microsoft.Extensions.Logging.Debug": "1.0.0",
        "Microsoft.Extensions.Options.ConfigurationExtensions": "1.0.0",
        "Microsoft.VisualStudio.Web.BrowserLink.Loader": "14.0.0",
        "Serilog.Extensions.Logging": "1.0.0",
        "Serilog.Sinks.RollingFile": "2.1.0"
    },
    "frameworks": {
        "net451": {}
    },
    "publishOptions": {
        "include": [
            "wwwroot",
            "Views",
            "Areas/**/Views",
            "appsettings.json",
            "appsettings.Development.json",
            "appsettings.Staging.json",
            "appsettings.Release.json"

        ]
    },
    "runtimeOptions": {
        "configProperties": {
            "System.GC.Server": true
        }
    },
    "scripts": {
        "prepublish": [ "npm install", "bower install", "gulp clean", "gulp min" ],
        "postpublish": [ "dotnet publish-iis --publish-folder %publish:OutputPath% --framework %publish:FullTargetFramework%" ]
    },
    "tools": {
        "BundlerMinifier.Core": "2.0.238",
        "Microsoft.AspNetCore.Razor.Tools": "1.0.0-preview2-final",
        "Microsoft.AspNetCore.Server.IISIntegration.Tools": "1.0.0-preview2-final"
    }
}
`

**Comment by ArgusDev** [comment]
It looks like a Microsoft bug (among of a lot of other ones). I'm not sure why nobody else experienced the same problem.
I can confirm that I managed to get it working by upgrading my Identity Server to ASP.NET Core 1.1.0-alpha1-21853.
Good luck.

**Comment by asteffes** [comment]
Thanks for your reply @ArgusDev. Follow-up question: can you please verify that in your setup you're targeting the alpha framework for .net core 1.1.0? Or, are you referring to a different release version of Identity Server 3 itself?

In my project I'm targeting the desktop .NET version in my project.json listed above: net451. I have other project dependencies for my Identity Server project that are also running net451, and as I understand it, I won't be able to reference from a .NET Core 1.x targeted project to "desktop" .NET frameworks.

**Comment by ArgusDev** [comment]
My project.json file has "1.1.0-alpha1-21853" version for all "Microsoft.AspNetCore.*", "Microsoft.Extensions.*" assemblies and "1.0.0-preview3-21853" for all tools. I'm using "IdentityServer3": "2.5.0".

I also target net451 only and I also reference a lot of old style net451 class libraries projects of the same solution without any problem (there are some tricks referencing third party assemblies or assemblies that are built as a part of another solution but it's still doable).

**Comment by leastprivilege** [comment]
we have a bug fix for identityserver that will work with v1.0 of aspnet core. Publishing today or tomorrow.

**Comment by ArgusDev** [comment]
I can confirm that IdentityServer3 2.5.2 works Ok with ASP.NET Core 1.0.0.
Thank you.
